File: Python/test_mbti.py
import mbti

import unittest
#from unittest import TestCase


class TestMbti(unittest.TestCase):

	
	def test_function1_exist(self):
		mbti.extrovert_introvert("ola")


	def test_function2_is_valid(self):
		actual = mbti.extrovert_introvert("ola")
		expected = "I"
		self.assertEqual(actual, expected)


	def test_function3_is_valid(self):
		actual = mbti.extrovert_introvert("ola")
		expected = "E"
		self.assertEqual(actual, expected)
	
	

	def test_function4_is_valid(self):
		actual = mbti.sensing_intuitive("ola")
		expected = "S"
		self.assertEqual(actual, expected)



	def test_function5_is_valid(self):
		actual = mbti.sensing_intuitive("ola")
		expected = "N"
		self.assertEqual(actual, expected)

	

	def test_function6_is_valid(self):
		actual = mbti.thinking_feeling("ola")
		expected = "T"
		self.assertEqual(actual, expected)



	def test_function7_is_valid(self):
		actual = mbti.thinking_feeling("ola")
		expected = "F"
		self.assertEqual(actual, expected)



	def test_function8_is_valid(self):
		actual = mbti.judging_percerptive("ola")
		expected = "J"
		self.assertEqual(actual, expected)



	def test_function9_is_valid(self):
		actual = mbti.judging_percerptive("ola")
		expected = "P"
		self.assertEqual(actual, expected)

File: Java/TestMBTI.java
public class TestMBTI {

	@Test
	public void testFunctionIsValid1() {
        	String actual = MBTI.extrovertIntrovert("ola");
		String expected = "I";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid2() {
        	String actual = MBTI.extrovertIntrovert("ola");
		String expected = "E";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid3() {
        	String actual = MBTI.sensingIntuitive("ola");
		String expected = "S";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid4() {
        	String actual = MBTI.sensingIntuitive("ola");
		String expected = "N";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid5() {
        	String actual = MBTI.thinkingFeeling("ola");
		String expected = "T";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid6() {
        	String actual = MBTI.thinkingFeeling("ola");
		String expected = "F";
		assertEquals(expected, actual);
	}


	@Test
	public void testFunctionIsValid7() {
        	String actual = MBTI.judgingPerceptive("ola");
		String expected = "J";
		assertEquals(expected, actual);
	}



	@Test
	public void testFunctionIsValid8() {
        	String actual = MBTI.judgingPerceptive("ola");
		String expected = "P";
		assertEquals(expected, actual);
	}

}